modules/rtm/R/invert.auto.R

```diff
@@ -67,11 +67,30 @@ invert.auto <- function(observed, invert.options, return.samples=TRUE, save.samp
     i.try <- 1
     while(try.again & i.try <= n.tries){
         print(sprintf("Attempt %d of %d", i.try, n.tries))
-        samps.list <- list()
-        for(chain in 1:nchains){
-            print(sprintf("Chain %d of %d", chain, nchains))
-            invert.options$inits <- inits.function() 
-            samps.list[[chain]] <- invert.custom(observed=observed, invert.options=invert.options, quiet=quiet)
+        if(parallel){
+            if(!require(parallel)) stop("'parallel' package not installed")
+            invert.function <- function(x){
+                invert.options$inits <- inits.function()
+                samps <- invert.custom(observed=observed, invert.options=invert.options, quiet=quiet)
+                return(samps)
+            }
+            if(is.null(parallel.cores)){
+                cl <- makeCluster(detectCores() - 1, "FORK")
+            } else {
+                if(!is.numeric(parallel.cores) | parallel.cores %% 1 != 0){
+                    stop("Invalid argument to 'parallel.cores'. Must be integer or NULL")
+                }
+                cl <- makeCluster(parallel.cores, "FORK")
```

Comment:
what happens if the user requests more cores than what is available? would this cause an error or just slow things down? would you like to throw a warning if the latter happens